spec/cancancan/model_adapters/neo4j_adapter_spec.rb

```diff
@@ -381,10 +372,15 @@
     it 'returns `not (condition)` for single `cannot` definition in front of default `can` condition' do
       @ability.can :read, Article
       @ability.cannot :read, Article, published: false, secret: true
-      cypher_string = 'WHERE ((true)) AND NOT((article.published=false)'\
-                      ' AND (article.secret=true))'
+      cypher_str = 'OPTIONAL MATCH (article_1:`Article`) WHERE (true) WITH '\
+        'collect(DISTINCT article_1) as article_1_col UNWIND article_1_col'\
+        ' as article_2 WITH DISTINCT article_2 as article_2 MATCH (article_2)'\
+        ' WHERE NOT(article_2.published = {article_2_published} AND '\
+        'article_2.secret = {article_2_secret}) WITH collect(DISTINCT'\
+        ' article_2) as article_2_col UNWIND article_2_col as article_can'\
+        ' WITH DISTINCT article_can as article'
```

Comment:
How about better formatting of the cypher like neo4j.rb does. It is nice to have but don't let this hold up the PR. But it looks like you might still not be reusing neo4j.rb code.

Reply:
Variable names, I am building them myself for the unwind and collect clauses, rest is nro4j.rb code.